### PR TITLE
[SMALLFIX] Fixing bug in URIUtils and other minor cleanup.

### PR DESCRIPTION
**Pull request by jsimsa**
Fixes a bug introduced by https://github.com/Alluxio/alluxio/pull/4862/files (`URIUtils#appendPath` would discard the port). This should fix the HDFS build.

This PR also includes minor cleanup of logging (following our logging style guideline).

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/cli/Format.java [hunk 1]
```diff
@@ -77,7 +84,7 @@ public static void main(String[] args) {
       System.exit(-1);
     }
     try {
-      format(args[0]);
+      format(Mode.valueOf(args[0]));
```
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/cli/Format.java]: we should convert args to uppercase, and double check that the error message is helpful when args[0] is something other than "master" or "worker"

Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/cli/Format.java]: done ... the error message is the same as before this PR

File: core/common/src/main/java/alluxio/util/URIUtils.java [hunk 2]
```diff
@@ -45,8 +45,8 @@ private URIUtils() {} // prevent instantiation
    * @throws URISyntaxException if URI syntax error is encountered
    */
   public static URI appendPath(URI base, String path) throws URISyntaxException {
-    return new URI(base.getScheme(), base.getHost(), PathUtils.concatPath(base.getPath(), path),
-        base.getFragment());
+    return new URI(base.getScheme(), base.getAuthority(),
+        PathUtils.concatPath(base.getPath(), path), base.getQuery(), base.getFragment());
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/util/URIUtils.java]: There's no utility which already does this?

Comment (jsimsa) [inline core/common/src/main/java/alluxio/util/URIUtils.java]: not to my knowledge

File: core/server/common/src/main/java/alluxio/cli/Format.java (hunk 1 above)
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/cli/Format.java]: I just tested and the error message changes to something like `java.lang.IllegalArgumentException: No enum constant alluxio.cli.Format.Mode.TEST`. We could catch exception here and print usage.

Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/cli/Format.java]: done

**Pull request merged**